File: api/schema.py
from pydantic import BaseModel
from typing import List, Optional


# General
class TimestampText(BaseModel):
    timestamp: str
    text: str

class TimestampTextList(BaseModel):
    items: List[TimestampText]

# Description
class DescriptionOptions(BaseModel):
    sample: Optional[bool] = False

class DescriptionRequest(BaseModel):
    url: str
    options: DescriptionOptions

# Commentary
class CommentaryOptions(BaseModel):
    intro: bool
    outro: bool
    temperature: float

class CommentaryRequest(BaseModel):
    items: TimestampTextList
    options: CommentaryOptions

# Audio
class AudioOptions(BaseModel):
    stability: float

class AudioResponse(BaseModel):
    items: List[str]

class AudioRequest(BaseModel):
    items: TimestampTextList
    options: AudioOptions

class VideoOptions(BaseModel):
    bw: bool
    playSound: bool
    subtitlesEnabled: bool
    subtitlesSize: int

class VideoRequest(BaseModel):
    options: VideoOptions

class VideoResponse(BaseModel):
    filename: str

File: ai-vid-gen/src/lib/schema.ts
export type TimestampText = {
	timestamp: string;
	text: string;
};

export type TimestampTextList = {
	items: TimestampText[];
};

export type DescriptionOptions = {
	sample?: boolean;
};

export type CommentaryOptions = {
	intro?: boolean;
	outro?: boolean;
	temperature?: number;
};

export type AudioOptions = {
	stability: number;
};

export type FileResponse = {
	items: string[];
};

export type VideoOptions = {
	bw: boolean;
	playSound: boolean;
	subtitlesEnabled: boolean;
	subtitlesSize: number;
};
